security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdClient.java

```diff
@@ -135,7 +135,7 @@ public Publisher<AuthenticationResponse> onCallback(HttpRequest<Map<String, Obje
             if (LOG.isTraceEnabled()) {
                 LOG.trace("Received an authorization error response from provider [{}]. Error: [{}]", getName(), errorResponse.getError());
             }
-            throw new AuthorizationErrorResponseException(errorResponse);
+            return Flux.error(new AuthorizationErrorResponseException(errorResponse));
```

Comment:
return the error as a reactor error so it can be handled in the Flux error handling [here](https://github.com/micronaut-projects/micronaut-security/pull/1219/files#diff-6ad660652004d4dec2ff97c6a0f27e77a75a44ada0ab18ffdf2a04b4bca289bbR89)